File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/question/QuestionDetailResponse.java
package org.example.olympic_ot_project.dto.exam.question;

import lombok.Data;
import org.example.olympic_ot_project.core.QuestionLevel;
import org.example.olympic_ot_project.core.QuestionType;
import org.example.olympic_ot_project.dto.exam.option.QuestionOptionResponse;

import java.util.List;

@Data
public class QuestionDetailResponse {

    private Integer id;
    private String content;
    private String answer;
    private Integer score;
    private String imageUrl;
    private String videoUrl;
    private QuestionType type;
    private QuestionLevel level;
    private Integer timeLimit;
    private Integer categoryId;
    private String categoryName;

    private List<QuestionOptionResponse> options;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/question/question_detail_response.dart
import 'package:olympic_it_project/core/question_level.dart';
import 'package:olympic_it_project/core/question_type.dart';
import 'package:olympic_it_project/dto/admin_manager/question/question_option_response.dart';

class QuestionDetailResponse {
  final int id;
  final String content;
  final String? answer;
  final int score;
  final String? imageUrl;
  final String? videoUrl;
  final QuestionType type;
  final QuestionLevel level;
  final int? categoryId;
  final String? categoryName;
  final List<QuestionOptionResponse> options;

  QuestionDetailResponse({
    required this.id,
    required this.content,
    required this.answer,
    required this.score,
    required this.imageUrl,
    required this.videoUrl,
    required this.type,
    required this.level,
    required this.categoryId,
    required this.categoryName,
    required this.options,
  });

  factory QuestionDetailResponse.fromJson(Map<String, dynamic> json) {
    return QuestionDetailResponse(
      id: json["id"],
      content: json["content"],
      answer: json["answer"],
      score: json["score"],
      imageUrl: json["imageUrl"],
      videoUrl: json["videoUrl"],
      type: QuestionType.values.firstWhere((e) => e.name == json["type"]),
      level: QuestionLevel.values.firstWhere((e) => e.name == json["level"]),
      categoryId: json["categoryId"],
      categoryName: json["categoryName"],
      options: (json["options"] as List? ?? [])
          .map((e) => QuestionOptionResponse.fromJson(e))
          .toList(),
    );
  }
}